### PR TITLE
Re-package `trino`, making sure to build every package

### PR DESCRIPTION
**Pull request by pdeslaur**
The previous packaging was only building the `:trino-server`, and caused plugins to be downloaded from Maven, instead of being built locally. This PR fixes and simplifies the build process.

We also bumped dependencies to fix CVE-2023-36478 and CVE-2023-5072

### CODE REVIEW

File: trino.yaml [hunk 1]
```diff
@@ -32,49 +32,36 @@ pipeline:
...
+      cp ./core/trino-server/target/trino-server-${{package.version}}/NOTICE ${{targets.destdir}}/usr/lib/trino/
+      cp ./core/trino-server/target/trino-server-${{package.version}}/README.txt ${{targets.destdir}}/usr/lib/trino/
+      cp -r ./core/trino-server/target/trino-server-${{package.version}}/bin ${{targets.destdir}}/usr/lib/trino/bin/
+      cp -r ./core/trino-server/target/trino-server-${{package.version}}/lib ${{targets.destdir}}/usr/lib/trino/lib/
 
-      rsync -av /home/build/trino-all/plugin/* /home/build/plugins/optional/
-      rm -rf /home/build/trino-all/plugin/*
-      rsync -av --ignore-existing /home/build/trino-all/* ${{targets.destdir}}/usr/lib/trino/
-      # rsync -av --ignore-existing /home/build/plugins/default/* ${{targets.destdir}}/usr/lib/trino/plugin/
+      # Copy default configuratoins.
```
Comment (jonjohnsonjr) [inline trino.yaml:52]:
```suggestion
      # Copy default configurations.
```

Comment (pdeslaur) [reply trino.yaml]: Dang it. Fixed!

File: trino.yaml [hunk 2]
```diff
@@ -32,49 +32,36 @@ pipeline:
...
-      # rsync -av --ignore-existing /home/build/plugins/default/* ${{targets.destdir}}/usr/lib/trino/plugin/
+      # Copy default configurations.
+      mkdir -p ${{targets.destdir}}/etc
+      cp -r core/docker/default/etc ${{targets.destdir}}/etc/trino
 
-      rsync -av --ignore-existing core/docker/default/etc/* ${{targets.destdir}}/etc/trino/
-      rsync -av --force core/docker/bin/* ${{targets.destdir}}/usr/lib/trino/bin/
-      rsync -av --ignore-existing client/trino-cli/target/trino-cli-${{package.version}}-executable.jar ${{targets.destdir}}/usr/bin/
-      rsync -av --ignore-existing /usr/lib/libjvmkill.so ${{targets.destdir}}/usr/lib/trino/bin/
+      # Copy docker helper scripts.
```
Comment (jonjohnsonjr) [inline trino.yaml:56]: I think we'll want to split some of this out into an oci-entrypoint style package e.g. https://github.com/wolfi-dev/os/blob/15c516cac756936e157b3e05166734e7b20fbfca/neo4j.yaml#L49

And a config package: https://wiki.inky.wtf/docs/teams/engineering/images/package-and-image-guides/serviceable-packages/

Comment (pdeslaur) [reply trino.yaml]: Done, great point!

**Review by jonjohnsonjr — APPROVED** [review]

**Pull request merged**